src/options/mkoptions.py

```diff
@@ -1235,122 +1319,50 @@ def parse_module(filename, file):
-                check_alias_attrib(filename, i, attrib, value)
-                if attrib in alias:
-                    perr(filename, i,
-                         "duplicate alias attribute '{}'".format(attrib))
-                assert alias is not None
-                alias[attrib] = value
-            # All other attributes are part of the module.
-            else:
-                if attrib in module:
-                    perr(filename, i,
-                         "duplicate module attribute '{}'".format(attrib))
-                check_module_attrib(filename, i, attrib, value)
-                module[attrib] = value
-        else:
-            perr(filename, i, "invalid attribute '{}'".format(line[0]))
-
-    # Save previously parsed option/alias
-    if option:
-        options.append(option)
-    if alias:
-        aliases.append(alias)
-
     # Check if parsed module attributes are valid and if all required
     # attributes are defined.
     check_attribs(filename, 1,
                   MODULE_ATTR_REQ, MODULE_ATTR_ALL, module, 'module')
     res = Module(module)
 
-    # Check parsed option/alias attributes and create option/alias objects and
-    # associate file name and line number with options/aliases (required for
-    # better error reporting).
-    assert len(option_lines) == len(options)
-    assert len(alias_lines) == len(aliases)
-    for i in range(len(options)):
-        attribs = options[i]
-        lineno = option_lines[i]
-        check_attribs(filename, lineno,
-                      OPTION_ATTR_REQ, OPTION_ATTR_ALL, attribs, 'option')
-        option = Option(attribs)
-        if option.short and not option.long:
-            perr(filename, lineno,
-                 "short option '{}' specified but no long option".format(
-                     option.short))
-        if option.type == 'bool' and option.handler:
-            perr(filename, lineno,
-                 'specifying handlers for options of type bool is not allowed')
-        if option.category != 'undocumented' and not option.help:
-            perr(filename, lineno,
-                 'help text is required for {} options'.format(option.category))
-        option.lineno = lineno
-        option.filename = filename
-        res.options.append(option)
-
-    for i in range(len(aliases)):
-        attribs = aliases[i]
-        lineno = alias_lines[i]
-        check_attribs(filename, lineno,
-                      ALIAS_ATTR_REQ, ALIAS_ATTR_ALL, attribs, 'alias')
-        alias = Alias(attribs)
-        alias.lineno = lineno
-        alias.filename = filename
-        res.aliases.append(alias)
+    if 'option' in module:
+        for attribs in module['option']:
+            lineno = 0
+            check_attribs(filename, lineno,
+                          OPTION_ATTR_REQ, OPTION_ATTR_ALL, attribs, 'option')
+            option = Option(attribs)
+            if option.mode and not option.help_mode:
+                perr(filename, lineno,
```

Comment:
IMHO we should change `perr` to take `lineno` as an optional argument (and not print out the line number in that case). No line number is better than a wrong one.

Reply:
Totally forgot about that argument. I actually wanted to change the error reporting since we don't have the line number information anymore. Now you can optionally pass an option/alias object from which a name/long option name is extracted for error reporting.